### PR TITLE
Adds support for 409 error code for grants

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/2428

UI implementation: https://github.com/brave/brave-ui/pull/326

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
sadly this is not easy to test as this is a real edge case scenario. What developers can do is hard code this response in bat_client.cc to see how it works.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: components/definitions/rewards.d.ts [hunk 1]
```diff
@@ -76,7 +76,7 @@ declare namespace Rewards {
     expiryTime: number
     captcha?: string
     hint?: string
-    status?: 'wrongPosition' | 'grantGone' | 'generalError' | number | null
+    status?: 'wrongPosition' | 'grantGone' | 'generalError' | 'grantAlreadyClaimed' | number | null
```
Comment (ryanml) [inline components/definitions/rewards.d.ts]: This is now getting pretty long, let's put this one in it's own type, `GrantStatus`, wdyt?

Comment (NejcZdovc) [reply components/definitions/rewards.d.ts]: done

File: package-lock.json [hunk 2]
```diff
@@ -463,7 +463,7 @@
     },
     "ansi-escapes": {
       "version": "3.1.0",
-      "resolved": "https://registry.npmjs.org/ansi-escapes/-/ansi-escapes-3.1.0.tgz",
+      "resolved": "http://registry.npmjs.org/ansi-escapes/-/ansi-escapes-3.1.0.tgz",
```
Comment (NejcZdovc) [inline package-lock.json]: Not sure why this happens, but it looks like npm bug https://npm.community/t/some-packages-have-dist-tarball-as-http-and-not-https/285/39
